From 790bf66035be3acf33488b88cc122660affa6999 Mon Sep 17 00:00:00 2001
From: Mark Asselstine <mark.asselstine@windriver.com>
Date: Tue, 15 Jul 2008 15:49:27 -0400
Subject: [PATCH] arm: ARM EABI socketcall

This patch make that the sys_socketcall system call is available when
EABI is used. The reason is that the user-space applications in both
glibc_std rootfs and glibc_small rootfs are still using sys_socketcall
to request kernel service.

Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 arch/arm/kernel/entry-common.S | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/kernel/entry-common.S b/arch/arm/kernel/entry-common.S
index f379c852dcb7..f22fd8af2d7b 100644
--- a/arch/arm/kernel/entry-common.S
+++ b/arch/arm/kernel/entry-common.S
@@ -353,7 +353,8 @@ ENTRY(\sym)
  * With EABI a couple syscalls are obsolete and defined as sys_ni_syscall.
  */
 	syscall_table_start sys_call_table
-#ifdef CONFIG_AEABI
+
+#if defined(CONFIG_AEABI) && !defined(CONFIG_OABI_COMPAT)
 #include <calls-eabi.S>
 #else
 #include <calls-oabi.S>
-- 
2.39.2

